pegasus/routes/hoc_routes.rb

```diff
@@ -25,10 +25,19 @@
 
   # set company to nil if not a valid company
   company = request.GET['company']
-  unless company.nil?
+  # Our partnership with cartoon network piggy-backs on our company profile.
+  # Pass their company param through to the congrats page even if no entry exists
+  # in the forms table.
+  unless company.nil? || company == 'CN'
     company = nil unless DB[:forms].where(kind: 'CompanyProfile', name: company).first
   end
 
+  # The lang parameter is used only by the cartoon network integration.
+  lang = request.GET['lang']
+  if company == 'CN' && lang == 'ar'
+    response.set_cookie('language_', {value: lang, domain: ".code.org", path: '/', expires: Time.now + (365*24*3600)})
+  end
+
   launch_tutorial(tutorial, company: company)
```

Comment:
launch tutorial sets the hour of code cookie and then redirects you to studio.code.org/s/<tutorial>/reset (for our tutorials), or to an external site.
```
def launch_tutorial(tutorial,params={})
  unless settings.read_only || hoc_activity_writes_disabled
    row = create_session_row(
      referer: request.referer_site_with_port,
      tutorial: tutorial[:code],
      company: params[:company],
      started_at: DateTime.now,
      started_ip: request.ip,
    )
    set_hour_of_code_cookie_for_row(row)
  end

  dont_cache
  redirect tutorial[:url], 302
end
```